universe/vikram.hello/index.js

```diff
@@ -0,0 +1,177 @@
+        }
+        if (sortURL) {
+            return Tp.Helpers.Http.get(sortURL).then((response) => {
+                let parsedResponse = JSON.parse(response);
+                return parsedResponse.results.map((result) => {
+                    const castQuery = "https://api.themoviedb.org/3/movie/" + String(result.id) + "/credits?api_key=654083fa9e049ea234e4ae94d3e65774&language=en-US";
+                    let id = new Tp.Value.Entity(String(result.id), result.title);
+                    return ({
+                        id,
+                        description: result.overview,
+                        release_date: new Date(result.release_date),
+                        rating_score: Number(result.vote_average),
+                    });
+                }).splice(0,3);
+            });
+        }
+        let query_term = '';
+        let searchType = 'movie';
+        if (hints && hints.filter) {
+            for (let [pname, op, value] of hints.filter) {
+                if (pname === 'id' && (op === '==' || op === '=~')) {
+                    if (value instanceof Tp.Value.Entity)
+                        query_term = encodeURIComponent(value);
+                }
+                else if (pname === 'actors' && (op === 'contains' || op === 'contains~')) {
+                    query_term = encodeURIComponent(value);
+                    searchType = 'actor';
+                }
+                
+            }
+        }
+        if (!query_term)
+            query_term = 'Avengers';
+        const multiQuery = tmdbAccess + multiSearch + this.constructor.metadata.auth.api_key + '&language=en-US&query=' + query_term + '&page=1%include_adult=false';
+        return Tp.Helpers.Http.get(multiQuery).then((response) => {
+            let parsedResponse = JSON.parse(response);
+            if (searchType == 'actor'){
+                console.log("Hello");
+            return parsedResponse.results[0].known_for.map((result) => {
+                    const castQuery = "https://api.themoviedb.org/3/movie/" + String(result.id) + "/credits?api_key=654083fa9e049ea234e4ae94d3e65774&language=en-US";
+                    return Tp.Helpers.Http.get(castQuery).then((response2) => {
+                        console.log(result2);
+                        let parsedCastQuery = JSON.parse(response2);
+                        return parsedCastQuery.cast.map((result2) => {
+                            let id = new Tp.Value.Entity(String(result.id), String(result.title));
+                            let oneDate = new Date(result.release_date);
+                            return ({
+                                id,
+                                actors: Array(result2.name),
+                                description: result.overview,
+                                release_date: oneDate,
+                                rating_score: Number(result.vote_average),
+                            });
+                        });
```

Comment:
This code is not correct. What you want is to put the list of actors in each movie object. Instead, you're returning an array of arrays of movie objects, each one of them has an `actors` array with only one actor.